File: medplum-java/medplum-java-model/src/main/java/com/fhir/BiologicallyDerivedProduct_Processing.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableBiologicallyDerivedProduct_Processing.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface BiologicallyDerivedProduct_Processing {
  Optional<CodeableConcept> procedure();

  Optional<List<Extension>> modifierExtension();

  Optional<String> timeDateTime();

  Optional<String> id();

  Optional<List<Extension>> extension();

  Optional<Period> timePeriod();

  Optional<String> description();

  Optional<Reference> additive();

  static ImmutableBiologicallyDerivedProduct_Processing.Builder builder() {
    return ImmutableBiologicallyDerivedProduct_Processing.builder();
  }
}

File: medplum-typescript/src/model/fhir/BiologicallyDerivedProduct_Processing.ts
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { Period } from "./Period";
import { Reference } from "./Reference";

export interface BiologicallyDerivedProduct_Processing {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    description: string | null | undefined;
    procedure: CodeableConcept | null | undefined;
    additive: Reference | null | undefined;
    timeDateTime: string | null | undefined;
    timePeriod: Period | null | undefined;
}
